src/rpc_server.rs

```diff
@@ -1123,6 +1135,21 @@ mod rpc_server_tests {
         Ok(())
     }
 
+    #[allow(clippy::shadow_unrelated)]
```

Comment:
nit: is this really needed?   I don't see any `let` shadowing.

Reply:
Sure, we can remove it.